lib/client.js

```diff
@@ -229,10 +243,28 @@ Client.prototype = {
   /// client.trigger('activation') // activating!
   /// ```
   trigger: function(name, data) {
-    if (!this._messageHandlers[name]) { return false; }
-    this._messageHandlers[name].forEach(function(handler) {
-      handler(data);
-    });
+    if (!this._messageHandlers[name] || repliesPending[name]) { return false; }
+    if (data && data.reply) {
+      repliesPending[name] = true;
+      // When receives a list of things to wrap as Promises,
```

Comment:
I wonder if this comment should go near the when definition rather than it's usage